src/layers/logging.rs

```diff
@@ -86,6 +94,140 @@ impl Layer for LoggingLayer {
+           match r {
+           Ok(r) => {
+               self.have_read += r;
+               debug!(target: "opendal::services", "service={} operation={} path={} size={:?} have_read={} poll read -> got: {}B", self.scheme, self.op, self.path, self.to_read, self.have_read, r);
+               Ok(r)
+           },
+               Err(e) => {
+                debug!(target: "opendal::services", "service={} operation={} path={} size={:?} have_read={} poll read -> failed: {:?}", self.scheme, self.op, self.path,  self.to_read, self.have_read, e);
+                   Err(e)
+            }
+        }
+        }).map_err(|e| {
+            error!(target: "opendal::services", "service={} operation={} path={} size={:?} have_read={} poll read -> errored: {:?}", self.scheme, self.op, self.path, self.to_read, self.have_read, e);
+            e
+        })
+    }
+
+    fn poll_read_vectored(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        bufs: &mut [IoSliceMut<'_>],
+    ) -> Poll<Result<usize>> {
+        Pin::new(&mut (*self.inner)).poll_read_vectored(cx, bufs).map(|r|{
+            match r {
+                Ok(r) => {
+                    self.have_read += r;
+                    debug!(target: "opendal::services", "service={} operation={} path={} size={:?} have_read={} poll vectored -> got: {:?}", self.scheme, self.op, self.path,  self.to_read, self.have_read, r);
+                    Ok(r)
+                },
+                Err(e) => {
+                    debug!(target: "opendal::services", "service={} operation={} path={} size={:?} have_read={} poll vectored -> failed: {:?}", self.scheme, self.op, self.path,  self.to_read, self.have_read, e);
+                    Err(e)
+                }
+            }
+        }).map_err(|e| {
+            error!(target: "opendal::services", "service={} operation={} path={} size={:?} have_read={} poll vectored -> errored: {:?}", self.scheme, Operation::Read, self.path, self.to_read, self.have_read, e);
+            e
+        })
+    }
+}
+
+pub struct LoggingStreamer {
+    scheme: Scheme,
+    path: String,
+    inner: DirStreamer,
+}
+
+impl LoggingStreamer {
+    pub(crate) fn new(scheme: Scheme, args: &OpList, streamer: Box<dyn DirStream>) -> Self {
+        Self {
+            scheme,
+            path: args.path().to_string(),
+            inner: streamer,
+        }
+    }
+}
+
+impl Stream for LoggingStreamer {
+    type Item = Result<DirEntry>;
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        Pin::new(&mut (*self.inner)).poll_next(cx).map(|o| {
+            if let Some(o) = o {
+                match o {
+                    Ok(dir) => {
+                        debug!(target: "opendal::service",
+                            "service={} operation={} path={} -> got entry, mode: {}, path: {}, content length: {:?}, last modified: {:?}, content md5: {:?}, etag: {:?}",
```

Comment:
Too verbose. We don't need to print all metadata in logging. `mode` and `path` is enough now.